web/src/compose_validate.js

```diff
@@ -368,27 +378,27 @@ function validate_stream_message_mentions(stream_id) {
                         "You do not have permission to use wildcard mentions in this stream.",
                 }),
                 compose_banner.CLASSNAMES.wildcards_not_allowed,
-                // Since we don't trigger validation in edit mode, the only place where
-                // wildcard warnings will appear is the compose banners container.
-                $("#compose_banners"),
+                $banner_container,
             );
             return false;
         }
 
-        if (user_acknowledged_wildcard === undefined || user_acknowledged_wildcard === false) {
+        if (
+            !is_wildcard_banner_shown_for_container.has(
+                $banner_container.closest("form").attr("id"),
+            )
+        ) {
             // user has not seen a warning message yet if undefined
```

Comment:
This comment is now wrong, I think?

It sounds like the logic here, though, is that we're now keeping track of whether we've shown the banner for this input in this session, rather than whether the user has acknowledged it; is the intention that if the user hides the banner by clicking the close button, we won't clear the state that we've shown the banner, and that's what we're achieving? I think that's correct, but want to make sure I understand correctly.

Reply:
Yeah, the comment is now wrong. 

> we're now keeping track of whether we've shown the banner for this input in this session, rather than whether the user has acknowledged it

that is right.

> is the intention that if the user hides the banner by clicking the close button, we won't clear the state that we've shown the banner, and that's what we're achieving?

Currently, this state is just keeping track of which banners are being shown to the user. While I was thinking about your question, I wondered if it was really necessary. If it's only doing such a simple job of tracking, there might be easier options to avoid adding multiple banners. Perhaps it's worth taking another look at `is_wildcard_banner_shown_for_container` in this iteration.

Reply:
I have given this more thought, and I believe you are correct. Currently, we are collecting the IDs of the inputs for which we have displayed the warning banner. We only remove the ID from the set after the message has been sent or saved.